oak_runtime/introspection_browser_client/components/Root/index.tsx

```diff
@@ -241,23 +241,52 @@ function useEvents() {
   );
   // Index of the event representing the last state change. Setting it lower
   // than the number of events enables inspecting past application states.
-  const [presentEventIndex, setPresentEventIndex] = React.useState<number>(
-    events.length - 1
-  );
+  const [presentEventIndex, setPresentEventIndex] = React.useState<number>(0);
+
+  const sessionStorageKeyRef = React.useRef<undefined | string>();
 
   React.useEffect(() => {
     async function loadEvents() {
       const serializedEvents: Uint8Array = await loadSerializedEvents();
       const events = introspectionEventsProto.Events.deserializeBinary(
         serializedEvents
       ).getEventsList();
+
+      if (events.length > 0) {
+        // Key the session storage to the first event, preventing state from
+        // being persisted across different inspected Oak applications.
+        sessionStorageKeyRef.current = `presentEventIndexSessionStorageKey-${events[0]
+          .getTimestamp()!
+          .getSeconds()}-${events[0].getTimestamp()!.getNanos()}`;
+
+        const sessionStorageValue = sessionStorage.getItem(
+          sessionStorageKeyRef.current
+        );
+
+        setPresentEventIndex(
+          sessionStorageValue
+            ? parseInt(sessionStorageValue)
+            : events.length - 1
+        );
+      }
+
       setEvents(events);
-      setPresentEventIndex(events.length - 1);
     }
 
     loadEvents();
   }, []);
 
+  // Persist the presentEventIndex across page refreshes & links.
+  // Ref: https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage
+  React.useEffect(() => {
```

Comment:
I don't really understand how these effects work in React, do you have any good reference I can read on?

Reply:
Good question! It's a complex topic, esp if you're new to React. 
The TL;DR that useEffect hooks are used to separate side effects from the rest of the component. It accepts to arguments a function (that performs a side effect) and a dependency array. Whenever an item in the dependency array changes, the effect reruns. 

I would recommend the react docs for an overview: https://reactjs.org/docs/hooks-effect.html
And https://overreacted.io/a-complete-guide-to-useeffect if you'd like to go deeper.